t/myriad.t

```diff
@@ -2,18 +2,138 @@ use strict;
 use warnings;
 
 use Myriad;
+use Myriad::Commands;
 use Test::More;
 use Test::Fatal;
+use Test::MockModule;
+use Test::MockObject;
+use Future::AsyncAwait;
+use IO::Async::Test;
 
-use Scalar::Util qw(refaddr);
+sub loop_notifiers {
+    my $loop = shift;
+
+    my @current_notifiers = $loop->notifiers;
+    my %loaded_in_loop = map { ref($_)  => 1} @current_notifiers;
+    return \%loaded_in_loop;
+}
+
+my $command_module = Test::MockModule->new('Myriad::Commands');
+my $command = 'test';
+my $command_is_called = 0;
+$command_module->mock($command, async sub { my ($self, $param) = @_; $command_is_called = $param; });
 
 my $myriad = new_ok('Myriad');
-isa_ok(my $f = $myriad->shutdown_future, 'Future');
-is(refaddr($f), refaddr($myriad->shutdown_future), 'same Future on multiple calls');
-is(exception {
-    $myriad->shutdown->get
-}, undef, 'can shut down without exceptions arising');
-is($f->state, 'done', 'shutdown future marked as done');
+my $metaclass = $myriad->META;
+
+subtest "class methods and proper initialization" => sub {
+    can_ok($myriad, $_) for qw(configure_from_argv loop registry redis rpc_client rpc http subscription storage add_service service_by_name ryu shutdown run);
+
+
+    my $command_param = 'Testing';
+    $myriad->configure_from_argv(('-l', 'debug', '--subscription_transport', 'perl', '--rpc_transport', 'perl', '--storage_transport', 'perl', $command, $command_param));
+
+    # Check configure_from_argv init objects
+    my $loop = $metaclass->get_slot('$loop')->value($myriad);
+    isa_ok($loop, 'IO::Async::Loop::Poll', 'Loop is set');
```

Comment:
doesn't need to be `::Poll` (preferably isn't - `Epoll` is generally a much better choice when there's more than a handful of handles to be watched)

```suggestion
    isa_ok($loop, 'IO::Async::Loop', 'Loop is set');
```